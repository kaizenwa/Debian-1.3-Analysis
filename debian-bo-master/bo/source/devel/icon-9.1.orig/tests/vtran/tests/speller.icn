procedure spell(n)
   local m
   n := integer(n) | stop(image(n)," is not an integer")
   if n <= 12 then return {
      "0zero,1one,2two,3three,4four,5five,6six,7seven,8eight,_
         9nine,10ten,11eleven,12twelve," ? {
            tab(find(n))
            move(*n)
            tab(upto(","))
            }
      }
   else if n <= 19 then return {
      spell(n[2] || "0") ?
         (if ="for" then "four" else tab(find("ty"))) || "teen"
      }
   else if n <= 99 then return {
      "2twen,3thir,4for,5fif,6six,7seven,8eigh,9nine," ? {
         tab(upto(n[1]))
         move(1)
         tab(upto(",")) || "ty" ||
            if n[2] ~= 0 then "-" || spell(n[2])
         }
      }
   else if n <= 999 then return {
      spell(n[1]) || " hundred" ||
         (if (m := n[2:0]) ~= 0 then " and " || spell(m) else "")
      }
   else if n <= 999999 then return {
      spell(n[1:-3]) || " thousand" ||
         (if (m := n[2:0]) ~= 0 then " and " || spell(m) else "")
      }
   else if n <= 999999999 then return {
      spell(n[1:-6]) || " million" ||
         (if (m := n[2:0]) ~= 0 then " and " || spell(m) else "")
      }
   else fail
end

procedure spellw(n)
   write(n, "	", spell(n))
   return
end

procedure main()
   every spellw(1 to 25)
   every spellw(30 to 110 by 3)
   spellw(945123342)
   every spellw(10000000 to 10000500 by 7)
end
