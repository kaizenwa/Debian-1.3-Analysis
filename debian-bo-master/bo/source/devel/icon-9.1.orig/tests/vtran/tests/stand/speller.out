
#line 0 "speller.icn"
procedure spell(n);
local m;
(n,((n)integer,((n)image," is not an integer")stop)|):=;
if (n,12)<= then return {
("0zero,1one,2two,3three,4four,5five,6six,7seven,8eight,9nine,10ten,11eleven,12twelve,",{
((n)find)tab;
((n)*)move;
((",")upto)tab
})?
} 

else if (n,19)<= then return {
(((n[2],"0")||)spell,((if ("for")= 
then "four" else (("ty")find)tab),"teen")||)?
} 
else if (n,99)<= then return {
("2twen,3thir,4for,5fif,6six,7seven,8eigh,9nine,",{
((n[1])upto)tab;
(1)move;
((((",")upto)tab,"ty")||,if (n[2],0)~= 
then ("-",(n[2])spell)||)||
})?
} 
else if (n,999)<= then return {
(((n[1])spell," hundred")||,(if (((m,n[2:0]):=),0)~= 
then (" and ",(m)spell)|| else ""))||
} 
else if (n,999999)<= then return {
(((n[1:(3)-])spell," thousand")||,(if (((m,n[2:0]):=),0)~= 
then (" and ",(m)spell)|| else ""))||
} 
else if (n,999999999)<= then return {
(((n[1:(6)-])spell," million")||,(if (((m,n[2:0]):=),0)~= 
then (" and ",(m)spell)|| else ""))||
} 
else fail;
end

procedure spellw(n);
(n,"	",(n)spell)write;
return ;
end

procedure main();
every (1 to 25)spellw;
every (30 to 110 by 3)spellw;
(945123342)spellw;
every (10000000 to 10000500 by 7)spellw;
end
